src/main/java/de/tum/in/www1/artemis/web/rest/dto/pageablesearch/CompetencyPageableSearchDTO.java

```diff
@@ -0,0 +1,56 @@
+package de.tum.in.www1.artemis.web.rest.dto.pageablesearch;
+
+import de.tum.in.www1.artemis.web.rest.dto.SearchResultPageDTO;
+
+/**
+ * Search DTO for a list of competencies matching the given search terms. The result should be paged,
+ * meaning that it only contains a predefined number of elements in order to not fetch and return too many.
+ *
+ * @see SearchResultPageDTO
+ */
+public class CompetencyPageableSearchDTO extends BasePageableSearchDTO<String> {
```

Comment:
Can we make this a record? I don't see why we would inherit from `CompetencyPagableSearchDTO` again in the future.

Reply:
This is not possible as Records cannot extend other classes (they already extend Record)

Reply:
Sorry, my bad. You're right 👍